### PR TITLE
Handle spawnDifficulties for serverside AreaTriggers

### PR DESCRIPTION
**Pull request by Traesh**
**Changes proposed:**

-  Add "SpawnDifficulties" column for serverside AreaTriggers
-  Useful for some dungeons where AT might be at different places depending of the difficulty (RP-triggering AT in Ulduman fe)

**Issues addressed:**

None

**Tests performed:**

Build, tested ingame

**Known issues and TODO list:**

None

### CODE REVIEW

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+UPDATE `areatrigger` SET `spawnDifficulties` = '1,2,23,8' WHERE `MapId` = 2293;
+UPDATE `areatrigger` SET `spawnDifficulties` = '14,15,16,17' WHERE `MapId` = 2296;
+UPDATE `areatrigger` SET `spawnDifficulties` = '15,14,16,17' WHERE `MapId` = 2296;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 2297;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 2364;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 2369;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 2370;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 2374;
+UPDATE `areatrigger` SET `spawnDifficulties` = '23,2,8' WHERE `MapId` = 2441;
+UPDATE `areatrigger` SET `spawnDifficulties` = ' 0' WHERE `MapId` = 2444;
```
Comment (mdX7) [inline sql/updates/world/master/9999_99_99_99_world.sql:261]: there are multiple lines with a prefixed space, likely copy pasta mistake. spawndifficulties shouldn't have spaces

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+UPDATE `areatrigger` SET `spawnDifficulties` = '14,15,16,17' WHERE `MapId` = 1520;
+UPDATE `areatrigger` SET `spawnDifficulties` = '14,15,16,17' WHERE `MapId` = 1530;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 1540;
+UPDATE `areatrigger` SET `spawnDifficulties` = '1,2,23' WHERE `MapId` = 1544;
+UPDATE `areatrigger` SET `spawnDifficulties` = '2,8,23' WHERE `MapId` = 1571;
+UPDATE `areatrigger` SET `spawnDifficulties` = '1,2,8,23' WHERE `MapId` = 1594;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 1637;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 1642;
+UPDATE `areatrigger` SET `spawnDifficulties` = '0' WHERE `MapId` = 1643;
+UPDATE `areatrigger` SET `spawnDifficulties` = '14, 15, 16, 17' WHERE `MapId` = 1648;
```
Comment (mdX7) [inline sql/updates/world/master/9999_99_99_99_world.sql:195]: also some lines with spaces in the middle

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,287 @@
+ALTER TABLE `areatrigger`
+    ADD COLUMN `spawnDifficulties` VARCHAR(100) NOT NULL DEFAULT '0' COLLATE 'utf8mb4_unicode_ci' AFTER `MapId`;
```
Comment (mdX7) [inline sql/updates/world/master/9999_99_99_99_world.sql:2]: UpperCamelCase is current standard for new fields

**Review by mdX7 — COMMENTED** [review]
great idea to add spawn difficulties :+1:

File: sql/updates/world/master/9999_99_99_99_world.sql (hunk 1 above)
Comment (Traesh) [reply sql/updates/world/master/9999_99_99_99_world.sql]: Done :)

File: sql/updates/world/master/9999_99_99_99_world.sql (hunk 2 above)
Comment (Traesh) [reply sql/updates/world/master/9999_99_99_99_world.sql]: Done

File: sql/updates/world/master/9999_99_99_99_world.sql (hunk 3 above)
Comment (Traesh) [reply sql/updates/world/master/9999_99_99_99_world.sql]: Done, it was already UCC in c++, fixed the SQL

**Review by mdX7 — APPROVED** [review]

**Pull request merged**